app/servicies/foreman_statistics/statistics/base.rb

```diff
@@ -0,0 +1,39 @@
+module ForemanStatistics
+  module Statistics
+    class Base
+      attr_reader :title, :count_by, :url
+
+      def initialize(options = {})
+        @id = options[:id]
+        @title = options[:title]
+        @search = options[:search]
+        @count_by = options[:count_by]
+        @organization_id = options[:organization_id]
+        @location_id = options[:location_id]
+        @url = options[:url] || build_url
+      end
+
+      def calculate
+        raise NotImplementedError, "Method 'calculate' method needs to be implemented"
+      end
+
+      def id
+        @id || count_by.to_s
+      end
+
+      def search
+        "/hosts?search=#{@search}"
```

Comment:
Maybe it was there before, but should this use the `url_for` helper?

Reply:
oh... for sure it should. It's even worth fixing in scope of this PR I guess